pkg/crc/machine/start.go

```diff
@@ -973,3 +979,17 @@ func startMicroshift(ctx context.Context, sshRunner *crcssh.Runner, ocConfig oc.
 
 	return nil
 }
+
+// The container.conf file has a setting called machine_enabled in the engine section,
+// which lets the podman client know that a command is running on an instance created
+// with the podman machine command. This allows the use of gvisor-tap-vsock when a
+// container is created with an exposed port. However, this setting should be disabled
+// for system mode networking, so that it doesn't prevent the creation of containers that
+// need to expose a port.
+// - https://github.com/crc-org/crc/issues/3515
+func removeContainerConfFile(sshRunner *crcssh.Runner) error {
+	if _, _, err := sshRunner.RunPrivileged("remove machine_enabled check", "rm -fr /etc/containers/containers.conf"); err != nil {
```

Comment:
If what we want to do is to set `machine_enabled` to false, why do we remove the whole file? It might even be possible to drop a file in ` /etc/containers/containers.conf.d/` to do this for us instead of changing `containers.conf`.

Note that this setting is deprecated (but still parsed) https://github.com/containers/common/blob/main/pkg/config/config.go#L343-L346

It uses a `/etc/containers/podman-machine` file now https://github.com/containers/common/blob/main/pkg/machine/machine.go
Maybe machine_enabled can always be set to false.

Reply:
>If what we want to do is to set machine_enabled to false, why do we remove the whole file? It might even be possible to drop a file in  /etc/containers/containers.conf.d/ to do this for us instead of changing containers.conf.

@cfergeau  There only one config in this file which we are creating as part of snc so  instead parsing and updating it or even dropping file to `/containers.conf.d` just to remove our changes in case system mode networking.

For microshift
https://github.com/crc-org/snc/blob/master/podman_changes.ks#L18-L21

For podman
https://github.com/crc-org/snc/blob/podman/fcos-config.yaml#L115-L121

> It uses a /etc/containers/podman-machine file now https://github.com/containers/common/blob/main/pkg/machine/machine.go

not aware of those changes but might be with latest podman, for microshift it is still 4.2.0 , in that case we need to make change in the snc side to have that marker file.

Reply:
>  There only one config in this file which we are creating as part of snc so instead parsing and updating it or even dropping file to /containers.conf.d just to remove our changes in case system mode networking.

And the day we want to add a new entry to this file, first it's going to break because we'll have forgotten about this, then we'll need to fix both snc and crc for something which should have been a snc-only change. Can we change our bundle creation process to create `/etc/containers/containers.conf.d/10-podman-machine-enabled.conf`, and then create/delete this file as needed?

Reply:
>Can we change our bundle creation process to create /etc/containers/containers.conf.d/10-podman-machine-enabled.conf, and then create/delete this file as needed?

Instead of that I will use the marker file which is part of https://github.com/containers/common/blob/main/pkg/machine/machine.go since the machine-enabled config is deprecated soon.

Reply:
> meanwhile we can use this hack (at least till bundle is updated with marker file) and then put another PR to remove the maker file as needed.

Fine with me!

>  At this moment this will atleast unblock microshift bundle on system mode networking side.

This could also be documented as a known issue, `podman run -p 8080:80` failing by default on linux with the microshift bundle is imo not blocking, but I understand others might have different opinions on this matter